x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -395,7 +428,10 @@ public void analyzedPlan(
             try {
                 // the order here is tricky - if the cluster has been filtered and later became unavailable,
                 // do we want to declare it successful or skipped? For now, unavailability takes precedence.
-                EsqlCCSUtils.updateExecutionInfoWithUnavailableClusters(executionInfo, result.indices.unavailableClusters());
+                EsqlCCSUtils.updateExecutionInfoWithUnavailableClusters(
```

Comment:
I see that `updateExecutionInfoWithUnavailableClusters` is always called on the result of `determineUnavailableRemoteClusters` - maybe we should put it inside?

Reply:
see https://github.com/elastic/elasticsearch/pull/130840/commits/2c1b04f4c714ce00611198cde54798a71401e094.